neuralpredictors/measures/np_functions.py

```diff
@@ -12,10 +14,157 @@ def corr(y1, y2, axis=-1, eps=1e-8, **kwargs):
     return (y1 * y2).mean(axis=axis, **kwargs)
+
+
+def oracle_corr_corrected(repeated_outputs):
+    """
+    Compute the corrected oracle correlations per neuron.
+
+    Args:
+        repeated_outputs (array-like): numpy array with shape (images, repeats, responses), or a list of lists of
+            responses (with len: num of repeats) per image.
+
+    Returns:
+        array: Corrected oracle correlations per neuron
+    """
+
+    if len(repeated_outputs.shape) == 3:
+        var_noise = repeated_outputs.var(axis=1).mean(0)
+        var_mean = repeated_outputs.mean(axis=1).var(0)
+    else:
+        var_noise, var_mean = [], []
+        for repeat in repeated_outputs:
+            var_noise.append(repeat.var(axis=0))
+            var_mean.append(repeat.mean(axis=0))
+        var_noise = np.mean(np.array(var_noise), axis=0)
+        var_mean = np.var(np.array(var_mean), axis=0)
+    return var_mean / np.sqrt(var_mean * (var_mean + var_noise))
+
+
+def oracle_corr(repeated_outputs):
+    """
+    Compute the oracle correlations per neuron.
+
+    Args:
+        repeated_outputs (array-like): numpy array with shape (images, repeats, responses), or a list of lists of
+            responses (with len: num of repeats) per image.
+
+    Returns:
+        array: Oracle correlations per neuron
+    """
+
+    if len(repeated_outputs.shape) == 3:
+        _, r, n = repeated_outputs.shape
+        oracles = (repeated_outputs.mean(axis=1, keepdims=True) - repeated_outputs / r) * r / (r - 1)
+        if np.any(np.isnan(oracles)):
+            logger.warning(
+                "{}% NaNs when calculating the oracle. NaNs will be set to Zero.".format(np.isnan(oracles).mean() * 100)
+            )
+        oracles[np.isnan(oracles)] = 0
+        return corr(oracles.reshape(-1, n), repeated_outputs.reshape(-1, n), axis=0)
+    else:
+        oracles = []
+        for outputs in repeated_outputs:
+            r, n = outputs.shape
+            # compute the mean over repeats, for each neuron
+            mu = outputs.mean(axis=0, keepdims=True)
+            # compute oracle predictor
+            oracle = (mu - outputs / r) * r / (r - 1)
+
+            if np.any(np.isnan(oracle)):
+                logger.warning(
+                    "{}% NaNs when calculating the oracle. NaNs will be set to Zero.".format(
+                        np.isnan(oracle).mean() * 100
+                    )
+                )
+                oracle[np.isnan(oracle)] = 0
+
+            oracles.append(oracle)
+        return corr(np.vstack(repeated_outputs), np.vstack(oracles), axis=0)
+
+
+def explainable_var(repeated_outputs, eps=1e-9):
+    """
+    Compute the explainable variance per neuron.
+
+    Args:
+        repeated_outputs (array): numpy array with shape (images, repeats, responses), or a list of lists of
+            responses (with len: num of repeats) per image.
+
+    Returns:
+        array: Corrected oracle correlations per neuron
+    """
+
+    ImgVariance = []
```

Comment:
variable name convention is not Pythonic here - use `total_var` instead of `TotalVar`, `img_variance` instead of `ImgVariance` and so on.